dashboard/src/components/AppView/AppView.tsx

```diff
@@ -100,19 +104,27 @@ class AppView extends React.Component<IAppViewProps, IAppViewState> {
       }, {});
     this.setState({ otherResources });
 
-    const deployments = manifest.filter(d => d.kind === "Deployment");
-    const services = manifest.filter(d => d.kind === "Service");
-    const ingresses = manifest.filter(d => d.kind === "Ingress");
     const sockets: WebSocket[] = [];
-    for (const d of deployments) {
-      sockets.push(this.getSocket("deployments", d.apiVersion, d.metadata.name, newApp.namespace));
-    }
-    for (const svc of services) {
-      sockets.push(this.getSocket("services", svc.apiVersion, svc.metadata.name, newApp.namespace));
-    }
-    for (const i of ingresses) {
-      sockets.push(this.getSocket("ingresses", i.apiVersion, i.metadata.name, newApp.namespace));
-    }
+    manifest.forEach(i => {
+      switch (i.kind) {
+        case "Deployment":
+          sockets.push(
+            this.getSocket("deployments", i.apiVersion, i.metadata.name, newApp.namespace),
+          );
+          break;
+        case "Service":
+          sockets.push(this.getSocket("services", i.apiVersion, i.metadata.name, newApp.namespace));
+          break;
+        case "Ingress":
+          sockets.push(
+            this.getSocket("ingresses", i.apiVersion, i.metadata.name, newApp.namespace),
+          );
+          break;
+        case "Secret":
+          Secret.get(i.metadata.name, newApp.namespace).then(sec => this.handleResource(sec));
```

Comment:
We shouldn't really be getting data this way, we should be getting this from the Redux state. Yes, the WebSockets are done this way, but it was done because handling persistent connections in the Redux state is more difficult to handle (need to appropriately close sockets when unmounting the component). I'm not a fan of it, and it needs to be refactored at some point (one idea is to have a component per WebSocket that we open, similar to the way https://www.npmjs.com/package/react-lazylog handles log streaming).

For this PR, we should change this to get the Secrets using a Redux action and receiving them as props, like we do with all other regular requests. This could be similar to how we get the Secrets for ServiceBindings.

Reply:
Following the same reasoning than here https://github.com/kubeapps/kubeapps/pull/855#pullrequestreview-180586454 I'd have just done the minimum to make this works for now. I agree with @andresmgot with taking the short path here mainly because we do not have a proper pattern in this section of the page to retrieve resources via redux yet.

Reply:
> mainly because we do not have a proper pattern in this section of the page to retrieve resources via redux yet.

Hmm, I disagree, we actually use redux to retrieve other resources, however this is the only place where we go through the manifest.